spec/tapioca/compilers/dsl/url_helpers_spec.rb

```diff
@@ -0,0 +1,293 @@
+        parlour = Parlour::RbiGenerator.new(sort_namespaces: true)
+        subject.decorate(parlour.root, Object.const_get(constant))
+        parlour.rbi
+      end
+    end
+
+    content = <<~RUBY
+      class Application < Rails::Application
+      end
+    RUBY
+
+    it("generates RBI for GeneratedPathHelpersModule") do
+      expected = <<~RUBY
+        # typed: strong
+        module GeneratedPathHelpersModule
+          include ActionDispatch::Routing::PolymorphicRoutes
+          include ActionDispatch::Routing::UrlFor
+        end
+      RUBY
+
+      assert_equal(expected, rbi_for(content, :GeneratedPathHelpersModule))
+    end
+
+    it("generates RBI for GeneratedUrlHelpersModule") do
+      expected = <<~RUBY
+        # typed: strong
+        module GeneratedUrlHelpersModule
+          include ActionDispatch::Routing::PolymorphicRoutes
+          include ActionDispatch::Routing::UrlFor
+        end
+      RUBY
+
+      assert_equal(expected, rbi_for(content, :GeneratedUrlHelpersModule))
+    end
+
+    it("generates RBI for GeneratedPathHelpersModule with helper methods") do
+      content = <<~RUBY
+        class Application < Rails::Application
+          routes.draw do
+            resource :index
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        module GeneratedPathHelpersModule
+          include ActionDispatch::Routing::PolymorphicRoutes
+          include ActionDispatch::Routing::UrlFor
+
+          sig { params(args: T.untyped).returns(String) }
+          def edit_index_path(*args); end
+
+          sig { params(args: T.untyped).returns(String) }
+          def index_path(*args); end
+
+          sig { params(args: T.untyped).returns(String) }
+          def new_index_path(*args); end
+        end
+      RUBY
+
+      assert_equal(expected, rbi_for(content, :GeneratedPathHelpersModule))
+    end
+
+    it("generates RBI for GeneratedUrlHelpersModule with helper methods") do
+      content = <<~RUBY
+        class Application < Rails::Application
+          routes.draw do
+            resource :index
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        module GeneratedUrlHelpersModule
+          include ActionDispatch::Routing::PolymorphicRoutes
+          include ActionDispatch::Routing::UrlFor
+
+          sig { params(args: T.untyped).returns(String) }
+          def edit_index_url(*args); end
+
+          sig { params(args: T.untyped).returns(String) }
+          def index_url(*args); end
+
+          sig { params(args: T.untyped).returns(String) }
+          def new_index_url(*args); end
+        end
+      RUBY
+
+      assert_equal(expected, rbi_for(content, :GeneratedUrlHelpersModule))
+    end
+
+    it("generates RBI for constant that includes path_helpers_module") do
+      content += <<~RUBY
+        class MyClass
+          include Rails.application.routes.named_routes.path_helpers_module
```

Comment:
Again, let's just concentrate on the `Rails.application.route.url_helpers` combinations.